### PR TITLE
Fix/use __typename when looking for other types

### PR DESCRIPTION
**Pull request by danstarns**
# Description

Supersedes #1200 

Changes look at the `__typename` field on the generated types over the 'top level type name`, this is because GraphQL Codegen is not preserving the type names. I have raised an issue about that here https://github.com/dotansimha/graphql-code-generator/issues/7714

Some refactors to the aggregation types - centrally storing them in the `Node` class. 

# Issue

#1130 

Closes: #1130

### CODE REVIEW

**Review by tbwiss — DISMISSED** [review]
Looks good!
Left a minor comment also for this one.

File: packages/ogm/tests/issues/1130.test.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+describe("issues/1130", () => {
+    test("should re-create issue and return types without throwing", async () => {
+        const typeDefs = `
+            type Company {
+              name: String!
+              industries: [SICIndustry!]! @relationship(type: "CONDUCTS_BUSINESS_IN", direction: OUT)
+            }
+            
+            # <-- Changing the name to "SicIndustry" works
```
[the remark this answers is not in the document]
Comment (danstarns) [reply packages/ogm/tests/issues/1130.test.ts:30]: I'll keep it just to copy exactly the reproduce steps!

File: packages/graphql/src/classes/Node.ts [hunk 2]
```diff
@@ -202,6 +207,13 @@ class Node extends GraphElement {
         };
     }
 
+    public get aggregateTypeNames(): AggregateTypeNames {
+        return {
+            selection: `${this.name}AggregateSelection`,
+            input: `${this.name}AggregateSelectionInput`,
```
Comment (darrellwarde) [inline packages/graphql/src/classes/Node.ts:213]: This should probably use the pascal case singular for consistency - for instance we would want `AWSInstance` to have `AwsInstanceAggregateSelection` like we do elsewhere... I think. 😬 

```suggestion
            selection: `${this.pascalCaseSingular}AggregateSelection`,
            input: `${this.pascalCaseSingular}AggregateSelectionInput`,
```

**Review by darrellwarde — COMMENTED** [review]
Almost LGTM!!! Just wondering whether we should be using the pascal case of the name rather than the name directly, like we do elsewhere.

File: packages/graphql/src/classes/Node.ts (hunk 2 above)
Comment (danstarns) [reply packages/graphql/src/classes/Node.ts:213]: We should however I think doing so could cause unforeseen breaking changes to users with unusual type names such as the one demonstrated in this issue `SICIndustry`.

Comment (darrellwarde) [reply packages/graphql/src/classes/Node.ts:213]: Shall we just do it? Thoughts @angrykoala and @tbwiss?

Comment (tbwiss) [reply packages/graphql/src/classes/Node.ts:213]: I vote for consistency in this case. so pro pascalCaseSingular. but yes we might get some bug reports due to this breaking change.

**Pull request merged**